DEV/org.openl.rules.util/src/org/openl/rules/util/Statistics.java

```diff
@@ -37,6 +43,471 @@ public void processNonNull(T value) {
+        return populationVariance != null ? Math.sqrt(populationVariance) : null;
+    }
+
+    public static BigDecimal stdevP(BigDecimal... values) {
+        BigDecimal populationVariance = varP(values);
+        return populationVariance != null ? populationVariance.sqrt(MathContext.DECIMAL128) : null;
+    }
+
+    public static BigDecimal stdevP(BigInteger... values) {
+        BigDecimal populationVariance = varP(values);
+        return populationVariance != null ? populationVariance.sqrt(MathContext.DECIMAL128) : null;
+    }
+
+    public static <T extends Number> Double varP(T... values) {
+        Double avg = Avg.avg(values);
+        if (avg == null) {
+            return null;
+        }
+        if (values.length == 1) {
+            return (double) 0;
+        }
+        return process(values, new Statistics.Result<>() {
+            @Override
+            public void processNonNull(T value) {
+                result = variance(value.doubleValue(), avg, result);
+            }
+
+            @Override
+            public Double result() {
+                return result == null ? null : result / counter;
+            }
+        });
+    }
+
+    public static BigDecimal varP(BigDecimal... values) {
+        BigDecimal avg = Avg.avg(values);
+        if (avg == null) {
+            return null;
+        }
+        if (values.length == 1) {
+            return BigDecimal.valueOf(0);
+        }
+        return process(values, new Statistics.Result<>() {
+            @Override
+            public void processNonNull(BigDecimal value) {
+                result = variance(value, avg, result);
+            }
+
+            @Override
+            public BigDecimal result() {
+                return result == null ? null : result.divide(BigDecimal.valueOf(counter), MathContext.DECIMAL128);
+            }
+        });
+    }
+
+    public static BigDecimal varP(BigInteger... values) {
+        BigDecimal avg = Avg.avg(values);
+        if (avg == null) {
+            return null;
+        }
+        if (values.length == 1) {
+            return BigDecimal.valueOf(0);
+        }
+        return process(values, new Statistics.Result<>() {
+            @Override
+            public void processNonNull(BigInteger value) {
+                result = variance(new BigDecimal(value), avg, result);
+            }
+
+            @Override
+            public BigDecimal result() {
+                return result == null ? null : result.divide(BigDecimal.valueOf(counter), MathContext.DECIMAL128);
+            }
+        });
+    }
+
+    public static <T extends Number> Double stdevS(T... values) {
+        Double sampleVariance = varS(values);
+        return sampleVariance == null ? null : Math.sqrt(sampleVariance);
+    }
+
+    public static BigDecimal stdevS(BigDecimal... values) {
+        BigDecimal sampleVariance = varS(values);
+        return sampleVariance == null ? null : sampleVariance.sqrt(MathContext.DECIMAL128);
+    }
+
```

Comment:
Please, organize public functions by classes to reduce size of Statistics class :
* StdDev (for S and P)
* Var (for S and P)
* Covar (for S and P)
* Corel (including rsq)
* Forecast
* Slope
* Intercept